src/game/server/entities/player.cpp

```diff
@@ -429,10 +432,11 @@ bool CBasePlayer::TakeDamage(entvars_t* pevInflictor, entvars_t* pevAttacker, fl
 			flArmor = pev->armorvalue;
 			flArmor *= (1 / flBonus);
 			flNew = flDamage - flArmor;
-			pev->armorvalue = 0;
+			if (!m_bInfiniteArmor)
+				pev->armorvalue = 0;
 		}
 		else
-			pev->armorvalue -= flArmor;
+			pev->armorvalue -= m_bInfiniteArmor ? pev->armorvalue : flArmor;
```

Comment:
The player will lose all armor when they take damage. This should be:
```cpp
else if (!m_bInfiniteArmor)
{
    pev->armorvalue -= flArmor;
}
```

Reply:
Oops, fixed.